lib/sbi_ipi.c

```diff
@@ -36,10 +36,21 @@ static int sbi_ipi_send(struct sbi_scratch *scratch, u32 hartid,
 	remote_scratch = sbi_hart_id_to_scratch(scratch, hartid);
 	if (event == SBI_IPI_EVENT_SFENCE_VMA ||
 	    event == SBI_IPI_EVENT_SFENCE_VMA_ASID) {
-		ipi_tlb_data = sbi_tlb_info_ptr(remote_scratch);
-		ipi_tlb_data->start = tlb_info->start;
-		ipi_tlb_data->size = tlb_info->size;
-		ipi_tlb_data->asid = tlb_info->asid;
+		ipi_tlb_fifo = sbi_tlb_info_fifo_ptr(remote_scratch);
+		ret = sbi_fifo_enqueue(ipi_tlb_fifo, data);
+		while (ret < 0) {
+			/**
+			 * For now, Busy loop until there is space in the fifo.
+			 * There may be case where target hart is also
+			 * enqueue in source hart's fifo. Both hart may busy
+			 * loop leading to a deadlock.
+			 * TODO: Introduce a wait/wakeup event mechansim to handle
+			 * this properly.
+			 */
+			__asm__ __volatile("nop");
+			__asm__ __volatile("nop");
+			ret = sbi_fifo_enqueue(ipi_tlb_fifo, data);
+		}
```

Comment:
Instead of duplicating the call to `sbi_fifo_enqueue()`, I think the while loop can be better written as:

```
while (sbi_fifo_enqueue(ipi_tlb_fifo, data) < 0) {
    __asm__ code
}
```

Reply:
Done.